search/397_match_only_text_doc_values_multiterm_search failing release tests

### CI Link

https://gradle-enterprise.elastic.co/s/djwhsfcygzctq/tests/overview?outcome=FAILED
https://gradle-enterprise.elastic.co/s/4khsfufbku3cu/tests/overview?outcome=FAILED

### Repro line

```text
./gradlew ":x-pack:qa:core-rest-tests-with-security:yamlRestTest" --tests "org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT.test {yaml=search/397_match_only_text_doc_values_multiterm_search/*}" -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key
```

### Does it reproduce?

Yes

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

```
[2026-06-01T18:36:29,938][INFO ][o.e.c.m.MetadataCreateIndexService] [test-cluster-0] failed on parsing mappings on index creation [test_mot_dv_low]org.elasticsearch.index.mapper.MapperParsingException: Failed to parse mapping: unknown parameter [index] on mapper [text_field] of type [match_only_text]
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:609)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:465)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.updateIndexMappingsAndBuildSortOrder(MetadataCreateIndexService.java:1790)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.lambda$applyCreateIndexWithTemporaryService$7(MetadataCreateIndexService.java:746)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.indices.IndicesService.withTempIndexService(IndicesService.java:780)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexWithTemporaryService(MetadataCreateIndexService.java:741)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequestWithV1Templates(MetadataCreateIndexService.java:891)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:684)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService.applyCreateIndexRequest(MetadataCreateIndexService.java:704)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.metadata.MetadataCreateIndexService$CreateIndexClusterStateUpdateTaskExecutor.execute(MetadataCreateIndexService.java:521)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService.innerExecuteTasks(MasterService.java:1240)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:1203)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService.executeAndPublishBatch(MasterService.java:364)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.lambda$run$2(MasterService.java:1989)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.action.ActionListener.run(ActionListener.java:490)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService$BatchingTaskQueue$Processor.run(MasterService.java:1986)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService$5.lambda$doRun$0(MasterService.java:1510)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.action.ActionListener.run(ActionListener.java:490)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.cluster.service.MasterService$5.doRun(MasterService.java:1488)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
    at java.base/java.lang.Thread.run(Thread.java:1516)
Caused by: org.elasticsearch.index.mapper.MapperParsingException: unknown parameter [index] on mapper [text_field] of type [match_only_text]
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.FieldMapper$Builder.parse(FieldMapper.java:1920)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:2079)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.FieldMapper$TypeParser.parse(FieldMapper.java:2029)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:678)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:586)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.RootObjectMapper.parse(RootObjectMapper.java:441)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.MappingParser.parseToBuilder(MappingParser.java:150)
    at org.elasticsearch.server@9.5.0/org.elasticsearch.index.mapper.MapperService.doMerge(MapperService.java:607)
    ... 23 more
```

--- a/qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/CcsCommonYamlTestSuiteIT.java
+++ b/qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/CcsCommonYamlTestSuiteIT.java
@@ -102,7 +102,8 @@ public class CcsCommonYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
         .setting("xpack.license.self_generated.type", "trial")
         .feature(FeatureFlag.TIME_SERIES_MODE)
         .feature(FeatureFlag.SYNTHETIC_VECTORS)
-        .feature(FeatureFlag.EXTENDED_DOC_VALUES_PARAMS);
+        .feature(FeatureFlag.EXTENDED_DOC_VALUES_PARAMS)
+        .feature(FeatureFlag.INDEX_DISABLED_BY_DEFAULT);
 
     private static ElasticsearchCluster remoteCluster = ElasticsearchCluster.local()
         .name(REMOTE_CLUSTER_NAME)

--- a/qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/RcsCcsCommonYamlTestSuiteIT.java
+++ b/qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/RcsCcsCommonYamlTestSuiteIT.java
@@ -102,6 +102,7 @@ public class RcsCcsCommonYamlTestSuiteIT extends ESClientYamlSuiteTestCase {
         .feature(FeatureFlag.TIME_SERIES_MODE)
         .feature(FeatureFlag.SYNTHETIC_VECTORS)
         .feature(FeatureFlag.EXTENDED_DOC_VALUES_PARAMS)
+        .feature(FeatureFlag.INDEX_DISABLED_BY_DEFAULT)
         .user("test_admin", "x-pack-test-password");
 
     private static ElasticsearchCluster fulfillingCluster = ElasticsearchCluster.local()

--- a/qa/smoke-test-multinode/src/yamlRestTest/java/org/elasticsearch/smoketest/SmokeTestMultiNodeClientYamlTestSuiteIT.java
+++ b/qa/smoke-test-multinode/src/yamlRestTest/java/org/elasticsearch/smoketest/SmokeTestMultiNodeClientYamlTestSuiteIT.java
@@ -39,6 +39,7 @@ public class SmokeTestMultiNodeClientYamlTestSuiteIT extends ESClientYamlSuiteTe
         .feature(FeatureFlag.RANDOM_SAMPLING)
         .feature(FeatureFlag.EXTENDED_DOC_VALUES_PARAMS)
         .feature(FeatureFlag.COLUMNAR_INDEX_MODE_FEATURE_FLAG)
+        .feature(FeatureFlag.INDEX_DISABLED_BY_DEFAULT)
         .build();
 
     public SmokeTestMultiNodeClientYamlTestSuiteIT(@Name("yaml") ClientYamlTestCandidate testCandidate) {

--- a/x-pack/qa/core-rest-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/CoreWithSecurityClientYamlTestSuiteIT.java
+++ b/x-pack/qa/core-rest-tests-with-security/src/yamlRestTest/java/org/elasticsearch/xpack/security/CoreWithSecurityClientYamlTestSuiteIT.java
@@ -53,6 +53,7 @@ public class CoreWithSecurityClientYamlTestSuiteIT extends ESClientYamlSuiteTest
         .feature(FeatureFlag.RANDOM_SAMPLING)
         .feature(FeatureFlag.EXTENDED_DOC_VALUES_PARAMS)
         .feature(FeatureFlag.COLUMNAR_INDEX_MODE_FEATURE_FLAG)
+        .feature(FeatureFlag.INDEX_DISABLED_BY_DEFAULT)
         .build();
 
     public CoreWithSecurityClientYamlTestSuiteIT(@Name("yaml") ClientYamlTestCandidate testCandidate) {

